hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializationTest.java

```diff
@@ -84,6 +87,64 @@ public void serializeAndDeserialize_IdentifiedDataSerializable_versionedFactory(
         assertEquals(person.name, deserialized.name);
     }
 
+    @Test
+    public void testClarifiedExceptionsForUnsupportedClassTypes() {
+        class LocalClass implements DataSerializable {
+            @Override
+            public void writeData(ObjectDataOutput out) {
+            }
+
+            @Override
+            public void readData(ObjectDataInput in) {
+            }
+        }
+
+        DataSerializable anonymousInstance = new DataSerializable() {
+            @Override
+            public void writeData(ObjectDataOutput out) {
+            }
+
+            @Override
+            public void readData(ObjectDataInput in) {
+            }
+        };
+
+        DataSerializable[] throwingInstances = {new LocalClass(), anonymousInstance, new NonStaticMemberClass()};
+
+        for (DataSerializable throwingInstance : throwingInstances) {
+            try {
+                ss.toObject(ss.toData(throwingInstance));
+            } catch (HazelcastSerializationException e) {
+                assertTrue(e.getCause() instanceof NoSuchMethodException);
```

Comment:
These `assertTrue()` calls should have a message including the class name, otherwise a failure will just produce an `AssertionError` with a line number, which is not nice to debug.

Also the `fail()` should include the class name.